system/jlib/jlog.hpp

```diff
@@ -1277,6 +1277,10 @@ interface jlib_decl IContextLogger : extends IInterface
     virtual IProperties * getClientHeaders() const = 0;
     virtual void setSpanAttribute(const char *name, const char *value) const = 0;
     virtual void setSpanAttribute(const char *name, __uint64 value) const = 0;
+    virtual void recordSpanException(IException * e, bool failed = true) const = 0;
```

Comment:
Are these useful?  Better to call
queryActiveSpan()->recordSpanException()...

The same argument could probably go for setSpanAttribute(), but I would avoid changing that until the span in the thread context question is settled.